File: Hashmap/2610.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        unordered_map<int, int> m;
        vector<vector<int>> res;

        for (int i : nums) m[i]++;
        while (m.size()) {
            vector<int> v;
            for (auto i : m) {                
                v.push_back(i.first);
            }
            for (int i : v) {
                if (!--m[i])
                    m.erase(i);
            }
            res.push_back(v);
        }
        return res;
    }
};

File: Hashmap/2610.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        cnt = Counter(nums)
        seen = set()
        res = []        
        
        while cnt:
            seen = set(cnt)
            res.append(list(seen))
            for i in list(seen):
                cnt[i] -= 1
                if cnt[i] == 0:
                    del cnt[i]                
            
        return res
